File: 2015/src/Task.ts
import { readFileSync } from "fs";

export abstract class Task {
    abstract partOne(): String;
    abstract partTwo(): String;

    protected readInput(suffix: string = "input"): string {
        const path = `./inputs/${this.constructor.name.toLowerCase()}_${suffix}.txt`
        return readFileSync(path, 'utf-8');
    }
}

File: 2022/src/main/kotlin/se/augustocesar/Task.kt
package se.augustocesar

abstract class Task {
    abstract fun partOne(): String
    abstract fun partTwo(): String

    protected fun readInput(suffix: String = "input"): String {
        val day = this.javaClass.name.split('.').last().lowercase()

        return this.javaClass.getResource("/inputs/${day}_${suffix}.txt")?.readText()
            ?: throw RuntimeException("Failed to load input")
    }
}
